docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md

```diff
@@ -438,6 +438,89 @@ spec:
         template: "{{ .cluster.name }}-{{ .machinePool.topologyName }}-{{ .random }}"
 ```
 
+### Defining a custom namespace for ClusterClass object
+
+As a user, I may need to create a `Cluster` from a `ClusterClass` object that exists only in a different namespace. To uniquely identify the `ClusterClass`, a `NamespaceName` ref is constructed from combination of:
+* `cluster.spec.topology.classNamespace` - namespace of the `ClusterClass` object.
+* `cluster.spec.topology.class` - name of the `ClusterClass` object.
+
+Example of the `Cluster` object with the `name/namespace` reference:
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: my-docker-cluster
+  namespace: default
+spec:
+  topology:
+    class: docker-clusterclass-v0.1.0
+    classNamespace: default
+    version: v1.22.4
+    controlPlane:
+      replicas: 3
+    workers:
+      machineDeployments:
+      - class: default-worker
+        name: md-0
+        replicas: 4
+        failureDomain: region
+```
+
+<aside class="note warning">
+
+<h1>Cluster rebase across namespaces</h1>
+
+Changing `classNamespace` is not supported in rebase procedure, while chanding `class` reference to a different `ClusterClass` from the same namespace performs regular `Cluster` rebase procedure.
```

Comment:
Might be I'm wrong, but I did not found a place where we are enforcing "Changing `classNamespace` is not supported".
Also, out of curiosity, why are we introducing this limitation?

Reply:
That’s mainly due to tests performed on that operation locally. It seems when changing `classNamespace`, many referenced template components change namespace too, as they have to preserve ownership references, and instead of `rebase` performing minimal modifications of the resource, it has to switch resources entirely. This seemed as a larger scope change.

I’ll add a validation on the field modifications.

Reply:
Thanks clarification
It would be great if we can add the same note as a godoc explanation for check above

Reply:
I composed a clarification message on the field. It seems there is no need for additional tests as `TestLocalObjectTemplatesAreCompatible` is already covering template namespace changes.